core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -7087,11 +7087,12 @@ private long setColumnAppendPosition(int columnIndex, long size, boolean doubleA
             if (columnType > 0) { // Not deleted
                 final long pos = size - getColumnTop(columnIndex);
                 if (ColumnType.isVarSize(columnType)) {
-                    dataSizeBytes = ColumnType.getDriver(columnType).setAppendPosition(
+                    ColumnTypeDriver driver = ColumnType.getDriver(columnType);
+                    dataSizeBytes = driver.setAppendPosition(
                             pos,
                             auxMem,
                             dataMem
-                    );
+                    ) - driver.getMinAuxVectorSize();
```

Comment:
nit: I don't think this compensation makes any good, but it's a minor.

Reply:
Motivation: If we don't compensate, then a record in an empty table with a single `string` column will be estimated as `8` bytes, regardless of other columns. 

If we compensate then it gives a chance to `TableUtils.estimateAvgRecordSize()` since it's guarded like this:

```java
    public long getPartitionO3SplitThreshold() {
        long splitMinSizeBytes = configuration.getPartitionO3SplitMinSize();
        return splitMinSizeBytes /
                (avgRecordSize != 0 ? avgRecordSize : (avgRecordSize = TableUtils.estimateAvgRecordSize(metadata)));
    }
```
or this: https://github.com/questdb/questdb/blob/5691a98eca1748d3e3856965c689b1149257c230/core/src/main/java/io/questdb/cairo/TableWriter.java#L4732-L4736

Alternatively, the contract of `setAppendPos()` could be that it returns only the size occupied by useful entries and it won't include the size used by artificial entries. but it again sounded too complicated. 

wdyt?

Reply:
see this: https://github.com/jerrinot/questdb/blob/5691a98eca1748d3e3856965c689b1149257c230/core/src/main/java/io/questdb/cairo/TableWriter.java#L7073-L7079

That's the only place where the return value is `driver.setAppendPos()` is used. There is a special case for 0 rows. I believe this place is *the* reason for the original behaviour where the driver.setAppendPos(0) returned 28.